src/main/java/knightminer/inspirations/common/item/HidableBlockItem.java

```diff
@@ -0,0 +1,38 @@
+package knightminer.inspirations.common.item;
+
+import knightminer.inspirations.common.IHidable;
+import net.minecraft.block.Block;
+import net.minecraft.item.BlockItem;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemGroup;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.NonNullList;
+
+import javax.annotation.Nonnull;
+import java.util.function.Supplier;
+
+public class HidableBlockItem extends BlockItem implements IHidable {
```

Comment:
Isn't this redundant? If I remember, BlockItem already redirects to the Block's `fillItemGroup` method, so you are just telling it not to show twice

Reply:
In that sense yes, but I'm wanting every item and block to implement IHidable so we can then hide/show them in JEI and the like.

Reply:
In that case, maybe keep the isHidden implementation and skip the fillItemGroup call as it will be handled by the block later on?